packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx

```diff
@@ -73,7 +94,16 @@ export const ProjectsScene: FC<ProjectSceneProps> = () => {
       </Space>
       <ProjectList />
       <Space pt="xlarge">
-        <Button iconBefore={<Add />} onClick={handleAdd} disabled={isLoading}>
+        {!!judgingStarts && (
+          <Span color={eraColor(judgingStarted ? Era.past : Era.future)}>
+            {judgingStarts}
+          </Span>
+        )}
+        <Button
+          iconBefore={<Add />}
+          onClick={handleAdd}
+          disabled={(isLoading || judgingStarted) && !canLockProject(hacker)}
+        >
           Add Project
         </Button>
         <>
```

Comment:
The line below this one already demonstrates dependence on hackathon and hacker props to render. Also do not be confused by useSelector. It is almost the same as mapState, which maps state to props on the component. If a prop changes, the component re-renders.

Reply:
Thanks for the explanation about `useSelector()`. I think there may be other places with `useEffect()` that don't need it, but we can worry about that when we switch to the new persistence layer. It is possible someone could be on the project page when they reload, and not move from it until after it successfully loads, but we have a "not found" scenario that already handles that. Less code is more.